### PR TITLE
Update

### PR DESCRIPTION
**Pull request by stephin-thomas**
Remove deprecated dependencies and replace with newer alternatives. Change rust edition and syntax to 2021 edition.

### CODE REVIEW

File: src/app.rs [hunk 1]
```diff
@@ -145,12 +149,13 @@ impl App {
         let pb_bytes = multi_pb.add(pb_bytes);
         multi_pb.set_move_cursor(true);
         let pb_done = Arc::new(Mutex::new(()));
-        let pb_done2 = pb_done.clone();
-        thread::spawn(move || {
-            let _locked = pb_done2.lock().unwrap();
-            multi_pb.join().expect("join");
-        });
-        
+        // let pb_done2 = pb_done.clone();
```
Comment (acidnik) [inline src/app.rs:152]: this doesn't seems right

**Review by acidnik — CHANGES_REQUESTED** [review]
thanks for the PR! looks good, but please remove unnecessary commented code

**Comment by noamraph** [comment]
@acidnik I want to try to add parallelization to ppcp (I'm on a slow NFS, and copying a lot of files is VERY slow). I wanted to see if I can help with this PR.

I added a commit to it, and opened another PR so it will be easier to view the changes: https://github.com/acidnik/ppcp/pull/8

Regarding the removal of pb_done: now there's no need, and you can't, call `.join()` of indicatif. So I think there's no reason to have the extra mutex at all. In my PR I just removed it.

WDYT?

**Comment by acidnik** [comment]
@noamraph I still don't see how parallelization would speed up your network, but if you implement it and prove that it is faster (by showing the before/after numbers) I will merge the PR

**Comment by noamraph** [comment]
> @noamraph I still don't see how parallelization would speed up your network, but if you implement it and prove that it is faster (by showing the before/after numbers) I will merge the PR

Great, thanks! I have already seen parallelization give a very large benefit, with `rclone --multi-thread-streams=100`. The reason is that the NFS where I work has a pretty significant latency, and a high throughput. For example (pretty realistic), I can copy a single large file at 100MB/s, but copying each file takes 10ms no matter its size. So when I have a directory structure with 1M files 1kB each, copying it sequentially takes 3 hours, but copying one 1GB file takes 10 seconds. Parallelization makes everything much less latency-sensitive - having 100 threads can reduce the 3 hours to 100 or 200 seconds.

**Pull request merged**